torchrl/collectors/llm_collector.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from tensordict import lazy_stack, TensorDictBase
+
+from torchrl.collectors import (
+    LocalWeightUpdaterBase,
+    RemoteWeightUpdaterBase,
+    SyncDataCollector,
+)
+from torchrl.data import ReplayBuffer
+
+
+class LLMCollector(SyncDataCollector):
+    """A simplified version of SyncDataCollector for LLM inference."""
+
+    def __init__(
+        self,
+        env,
+        policy_factory,
+        *,
+        steps_per_batch: int = -1,
```

Comment:
```suggestion
        steps_per_batch: int,
```
Right now this is a value that we require. Long-term, when we'll go full async, we'll allow people to not specify anything because the output data will be written the buffer as soon as the trajectory is completed.